testutil/testkeyring/accounts.go

```diff
@@ -1,21 +1,36 @@
 package testkeyring
 
 import (
+	"context"
 	"encoding/base64"
 	"encoding/json"
 	"sync/atomic"
-
-	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/stretchr/testify/require"
 )
 
 // PreGeneratedAccount holds the address and mnemonic of an account which was
 // pre-generated by the `gen_accounts` package. It is intended to be used ONLY
 // in tests. It is useful for scenarios where it is necessary to know the address
 // of a specific key in a separate context from that of keyring and/or module
 // genesis state construction.
+//
+// TODO_BUG(@bryanchriswhite): investigate why the PreGeneratedAccount#Address seems to
```

Comment:
From a quick investigation, it seems like the contexts are different:

```go
// GetKeyring returns the cosmos-sdk client Keyring associated with the transaction factory.
func (txCtx cosmosTxContext) GetKeyring() cosmoskeyring.Keyring {
	return txCtx.txFactory.Keybase()
}
```

Can't tell where/how pre-generated accounts integrations with the txFactory.

Reply:
Can you spent an extra 15-30 minutes verifying this isn'tthe issue? Leave the TODO as if if you can't find it.

Reply:
> Can't tell where/how pre-generated accounts integrations with the txFactory.

It doesn't. I'm just using an in-memory keyring everywhere. See [gen.go](https://github.com/pokt-network/poktroll/blob/main/testutil/testkeyring/gen_accounts/gen.go#L35).

I think I figured out the issue though. There were two `string` arguments which were out of order in the call to `keyring.Keyring#NewAccount()`, one of which is the HD path, which would explain the discrepency:

Had:
```go
	_, err := keyRing.NewAccount(
		uid,
		pga.Mnemonic,
		keyring.DefaultBIP39Passphrase,
		cosmostypes.FullFundraiserPath,
		hd.Secp256k1,
	)
```

Needed:
```go
	_, err := keyRing.NewAccount(
		uid,
		pga.Mnemonic,
		cosmostypes.FullFundraiserPath,
		keyring.DefaultBIP39Passphrase,
		hd.Secp256k1,
	)
```